src/XrdCms/XrdCmsCluster.cc

```diff
@@ -1816,12 +1833,29 @@ XrdCmsNode *XrdCmsCluster::SelbyLoad(SMask_t mask, XrdCmsSelector &selR)
                       }
                    Multi = true;
                   }
+               }
           }
-
+    if (Config.P_randlb==1){ 
+    // pick a random weighed node
+    //
+    std::random_device rand_dev;
+    std::mt19937 generator(rand_dev());
+    std::uniform_int_distribution<int> distr(randomSel,selCap);
+    randomSel = distr(generator);
+    for(int i=0;i<=STHi;i++){
+      if(randomSel<=weighed[i]){
+        sp=NodeTab[i];
+        break;
+      }
+    }
+    }
+   delete [] weighed;
```

Comment:
Given that allocation only occurs when needed this delete can now move into the scope of the if statement,

Reply:
thanks, included in the second commit